elasticdl/python/elasticdl/image_builder.py

```diff
@@ -81,6 +79,55 @@ def build_and_push_docker_image(
     return image_name
 
 
+def remove_images(
+    docker_image_repository="",
+    docker_base_url="unix://var/run/docker.sock",
+    docker_tlscert="",
+    docker_tlskey="",
+):
+    """Remove all docker images with repository name equal to
+    docker_image_repository. If docker_image_repository is empty, it
+    will remove all images.
+    """
+    client = _get_docker_client(
+        docker_base_url=docker_base_url,
+        docker_tlscert=docker_tlscert,
+        docker_tlskey=docker_tlskey,
+    )
+    # Use repository tags to delete images
+    images = client.images(name=docker_image_repository, quiet=False)
+    for image in images:
+        repo_tags = image.get("RepoTags") or []
+        for repo_tag in repo_tags:
+            if repo_tag == "<none>:<none>":
+                # A special case where both repository and tag are none,
+                # and we need to delete it through Id in the following code.
+                continue
+            logger.info("Removing image %s" % repo_tag)
+            try:
+                client.remove_image(repo_tag)
+            except docker.errors.APIError as e:
+                logger.warning("Failed to delete image %s: %s" % (repo_tag, e))
+    # For image not having full repository tags, use ID instead.
+    # Note that, here we need to re-list images
+    images = client.images(name=docker_image_repository, quiet=False)
+    for image in images:
+        image_id = image.get("Id")
+        if image_id:
+            logger.info("Removing image %s" % image_id)
+            try:
+                client.remove_image(image_id)
+            except docker.errors.APIError as e:
+                logger.warning("Failed to delete image %s: %s" % (image_id, e))
+    # If removing all images, we also run prune to remove untagged images
+    if not docker_image_repository:
```

Comment:
Also check `--all`?

Reply:
`docker_image_repository` = "" actually means removing `all`, in the docker python API.